### ISSUE TITLE
The depth smooth loss goes to zero

### ISSUE THREAD

**Issue by feipan664** [bolianchen/pytorch_depth_from_videos_in_the_wild#8]
Dear Bolian, thank you for providing this code. While I was training on KITTI, the depth goes to zero. This is the screen cut of my trainig results.
![Snipaste_2022-04-30_09-36-11](https://user-images.githubusercontent.com/18341349/166083709-abf773f9-4e36-40cc-b839-e456b5485f57.png)
![Snipaste_2022-04-30_09-36-47](https://user-images.githubusercontent.com/18341349/166083715-232974c4-38f4-4a1d-8924-a61191245752.png)

Do you have any idea about it?

Regards,

Fei

**Comment by bolianchen** [comment]
@feipan664 Dear Feipan,

I just launched trainings on KITTI to find clues about the reported issue. I will give you feedback soon.

Best,
Bolian

**Comment by feipan664** [comment]
Thank you for your quick reply. I have set up the [depth smooth loss](https://github.com/bolianchen/pytorch_depth_from_videos_in_the_wild/blob/e8606013cec901da3a97948932f96f5689ea03dc/options/wild_options.py#L152) with 1e-3, and re-started the training. Now after 1 epoch, I can get this disparity map:
![Snipaste_2022-04-30_15-22-36](https://user-images.githubusercontent.com/18341349/166094422-82892938-fbb4-4f39-b12f-9b6e077befff.png)

**Comment by bolianchen** [comment]
@feipan664 Thank you very much for the feedback. It is a very good reference.    

I tried trainings with --learn_intrinsics option being activated and not and couldn't reproduce your results. If you followed [the code](https://github.com/bolianchen/pytorch_depth_from_videos_in_the_wild#generate-training-dataset)  to generate the training data, I guess the optimal hyper-parameter value might be affected by the difference in GPUs or PyTorch versions.  I tried (pytorch 1.5 + torchvision 0.6 + rtx2080Ti) and (pytorch 1.7.1 + torch 0.8.2 + rtx2070) and was able to produce depth while keeping smooth loss weight as 1e-2.
<pre>
python train.py --data_path $FULL_KITTI  \
                --png  --overwrite_old --boxify \
                --learn_intrinsics \
                --model_name "20220430/full_kitti_learn_cam"
</pre>
<img width="739" alt="Screen Shot 2022-04-30 at 10 16 17 PM" src="https://user-images.githubusercontent.com/39755828/166109274-2a4def08-9ee1-4752-beda-66c24c96578f.png">
<img width="344" alt="Screen Shot 2022-04-30 at 10 21 37 PM" src="https://user-images.githubusercontent.com/39755828/166109440-18d5da11-ad4f-4cb3-95a8-3e3b38b8bd75.png">

<pre>
python train.py --data_path $FULL_KITTI  \
                --png  --overwrite_old --boxify \
                --model_name "20220430/full_kitti_learn_cam"
</pre>
<img width="736" alt="Screen Shot 2022-04-30 at 10 24 50 PM" src="https://user-images.githubusercontent.com/39755828/166109551-4121bf32-8555-45a7-b3e1-944f5c38a89a.png">
<img width="343" alt="Screen Shot 2022-04-30 at 10 25 47 PM" src="https://user-images.githubusercontent.com/39755828/166109571-945efae0-43d2-4c8c-a4dc-f96a10471374.png">

**Comment by feipan664** [comment]
Thank you so much for the help. I follow your data generation code to prepare my KITTI data. My environment is pytorch1.7.0 + torchvision0.8.1 + RTX 2080 Ti. With smooth loss weight set up to 1e-3, I got these results so far:

![Snipaste_2022-05-01_00-05-26](https://user-images.githubusercontent.com/18341349/166111305-b115f90a-8106-45a1-b70b-c20ec90396df.png)
![Snipaste_2022-05-01_00-04-44](https://user-images.githubusercontent.com/18341349/166111311-36c2f2ba-88cd-4bc6-a0b9-a155fc40304b.png)
![Snipaste_2022-05-01_00-04-36](https://user-images.githubusercontent.com/18341349/166111312-71e063d4-fa0c-48b6-b43c-b008d037713f.png)

I would like to upgrade my PyTorch ver and run it again. 

Thank you again, Bolian!
